### PR TITLE
feat: Validate that resource maps to URI path for AIP-133 

### PR DESCRIPTION
**Pull request by acamadeo**
Part of #804

### CODE REVIEW

File: rules/aip0133/http_uri_resource.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+			}
+		}
+
+		return problems
+	},
+}
+
+func getURIPath(m *desc.MethodDescriptor) string {
+	for _, httpRule := range utils.GetHTTPRules(m) {
+		if httpRule.Method == "POST" {
```
[the remark this answers is not in the document]
Comment (acamadeo) [reply rules/aip0133/http_uri_resource.go]: Removed

File: rules/aip0133/http_uri_resource.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+					Descriptor: m.GetOutputType(),
+				})
+			}
+		}
+
+		return problems
+	},
+}
+
+func getURIPath(m *desc.MethodDescriptor) string {
```
[the remark this answers is not in the document]
Comment (acamadeo) [reply rules/aip0133/http_uri_resource.go]: Done

File: rules/aip0133/http_uri_resource.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+	"github.com/googleapis/api-linter/rules/internal/utils"
+	"github.com/jhump/protoreflect/desc"
+)
+
+// The resource name used in the Create method's URI should match the name used
+// in the resource definition.
+var httpURIResource = &lint.MethodRule{
+	Name: lint.NewRuleName(133, "http-uri-resource"),
+	OnlyIf: func(m *desc.MethodDescriptor) bool {
+		return isCreateMethod(m)
```
[the remark this answers is not in the document]
Comment (acamadeo) [reply rules/aip0133/http_uri_resource.go]: Done

File: rules/aip0133/http_uri_resource.go [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+	},
+	LintMethod: func(m *desc.MethodDescriptor) []lint.Problem {
+		problems := []lint.Problem{}
+
+		// Extract the suffix of the URI path as the collection identifier.
+		uriParts := strings.Split(getURIPath(m), "/")
+		collectionName := uriParts[len(uriParts)-1]
+
+		// Ensure that a collection identifier is provided.
+		if collectionName == "" {
```
[the remark this answers is not in the document]
Comment (acamadeo) [reply rules/aip0133/http_uri_resource.go]: Ack

**Review by noahdietz — APPROVED** [review]
Great stuff, thank you @acamadeo

**Pull request merged**